src/restful/middlewares/index.js

```diff
@@ -12,25 +12,22 @@ const protect = async (req, res, next) => {
       const decoded = jwt.verify(token, process.env.JWT_SECRET);
 
       const user = await UserService.findUserById(decoded.id);
-      if (!user.isLoggedIn)
-        return res
-          .status(401)
-          .json({ message: "Not Authorized, Not logged in" });
 
       if (!user.isActivated)
         return res
           .status(401)
-          .json({ message: "Not Authorized,User Account Is Not activated" });
+          .json({ error: "Not Authorized,User Account Is Not activated" });
 
       req.user = user;
       next();
     } catch (error) {
       console.log(error);
-      res.status(401).json({ message: "Not Authorized!" });
+      return res.status(401).json({ error: "Not Authorized!" });
     }
   }
+
   if (!token) {
-    res.status(401).json({ message: "Not Authenticated!" });
+    return res.status(401).json({ error: "Not Authenticated!" });
```

Comment:
@Patricknshimiyimana what do you think if we put this logic above try_catch in order to reduce redundant checks.

Hint:
```javascript
if (!token) {
  return res.status(401).json({ error: "Not Authenticated!" });
}

try {
  // rest content ...

} catch (error) {
  console.log(error);
  return res.status(401).json({ error: "Not Authorized!" });
}

```

Reply:
Hello @nicele08 
That try-catch block is wrapped in an if statement that checks for authorization headers,
And this code runs if there is no token, which would never be the case if place it where you are suggesting.
Check the screenshot below 👇
``` 
if (!token) {
  return res.status(401).json({ error: "Not Authenticated!" });
} 
``` 

![image](https://user-images.githubusercontent.com/73250459/236637619-d9d93bc2-6e22-40bd-90b3-e325075d5c58.png)

Reply:
> @Patricknshimiyimana what do you think if we put this logic above try_catch in order to reduce redundant checks.
> 
> Hint:
> 
> ```js
> if (!token) {
>   return res.status(401).json({ error: "Not Authenticated!" });
> }
> 
> try {
>   // rest content ...
> 
> } catch (error) {
>   console.log(error);
>   return res.status(401).json({ error: "Not Authorized!" });
> }
> ```

Hello @nicele08 
That try-catch block is wrapped in an if statement that checks for authorization headers,
And this code runs if there is no token, which would never be the case if place it where you are suggesting.
Check the screenshot below 👇
``` 
if (!token) {
  return res.status(401).json({ error: "Not Authenticated!" });
} 
``` 

![image](https://user-images.githubusercontent.com/73250459/236637619-d9d93bc2-6e22-40bd-90b3-e325075d5c58.png)

Reply:
@nicele08  That try-catch block is wrapped in an if statement that checks for authorization headers,
And this code runs if there is no token, which would never be the case if place it where you are suggesting.
You gerrit? Check the screenshot below 👇
``` 
if (!token) {
  return res.status(401).json({ error: "Not Authenticated!" });
} 
``` 
![image](https://user-images.githubusercontent.com/73250459/236639885-b69ea758-abc2-429f-861d-883dbd581dda.png)